internal/config/global_config.go

```diff
@@ -0,0 +1,117 @@
+package config
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/user"
+	"path"
+
+	"github.com/commitdev/zero/configs"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var GetCredentialsPath = getCredentialsPath
+
+type Projects map[string]*Project
```

Comment:
Is there a reason for these to be pointers to structs instead of just structs?

Reply:
Also can we call this something more specific? `Credentials` and `ProjectCredentials`?

Reply:
yeah, I need to persist the changes I make to the Project,  the map[string]Project doesn't get the updated value otherwise

Reply:
Only because `Save` is a pointer receiver. It could also be called like `config.Save(projects)` instead of `projects.Save()` There are a few things in this PR which look like they are structured in an OOP way which is not how things are typically done in Go.